cpp/Dockerfile

```diff
@@ -26,7 +26,6 @@ RUN export DEBIAN_FRONTEND=noninteractive && \
         g++ \
         gcc \
         git \
-        ninja-build \
```

Comment:
why this change?

Reply:
I think that's a holdover from trying to run integration tests in docker; at one point I got `ninja: command not found`. It's not relevant here so I'll remove it

Reply:
@emkornfield ninja is installed [in `ci/conda_env_cpp.yml`](https://github.com/apache/arrow/blob/a4dad3277f32b79e7da2488846f68e1987400786/ci/conda_env_cpp.yml#L33) so isn't this unused?